pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/immutable/ImmutableSegmentImpl.java

```diff
@@ -173,6 +175,16 @@ private File getValidDocIdsSnapshotFile() {
         V1Constants.VALID_DOC_IDS_SNAPSHOT_FILE_NAME);
   }
 
+  public boolean needReloadSegment(String segmentName, String crc, IndexLoadingConfig indexLoadingConfig)
```

Comment:
Seems `crc` is not needed?

Reply:
Suggest renaming it to `isReloadNeeded()`

Reply:
Done.